src/neural_mesh_simplification/losses/combined_loss.py

```diff
@@ -11,13 +12,18 @@
 
 class CombinedMeshSimplificationLoss(nn.Module):
     def __init__(
-        self, lambda_c: float = 1.0, lambda_e: float = 1.0, lambda_o: float = 1.0
+        self,
+        lambda_c: float = 1.0,
+        lambda_e: float = 1.0,
+        lambda_o: float = 1.0,
+        device=device("cpu")
     ):
         super().__init__()
-        self.prob_chamfer_loss = ProbabilisticChamferDistanceLoss()
-        self.prob_surface_loss = ProbabilisticSurfaceDistanceLoss()
-        self.collision_loss = TriangleCollisionLoss()
-        self.edge_crossing_loss = EdgeCrossingLoss()
+        self.device = device
+        self.prob_chamfer_loss = ProbabilisticChamferDistanceLoss().to(self.device)
+        self.prob_surface_loss = ProbabilisticSurfaceDistanceLoss().to(self.device)
+        self.collision_loss = TriangleCollisionLoss().to(self.device)
+        self.edge_crossing_loss = EdgeCrossingLoss().to(self.device)
         self.overlapping_triangles_loss = OverlappingTrianglesLoss()
```

Comment:
Is it intentional not to move this loss function to the same device?

```suggestion
        self.overlapping_triangles_loss = OverlappingTrianglesLoss().to(self.device)
```

Reply:
It's an oversight. Good catch!